### PR TITLE
Client: custom hook for upvote downvote

### PR DESCRIPTION
**Pull request by EugeneTeu**
To replicate the logic for answers. can just import and use this hook in any component that requires upvote downvote. Just remember to follow the props and pass the necessary API calls

### CODE REVIEW

**Review by AaronCQL — DISMISSED** [review]
LGTM

2 future points of consideration:

- Can make the `useUpvoteDownvote` hook support answers vote as well - so we can reuse it for answers
- For the UI, can the we merge the counter display together with the vote button so that it's one thing? Right now a little confusing because there is a button to upvote and a display of the votes itself.

File: client/utils/hooks/useUpvoteDownvote.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  qid,
+  checkVoteStatus,
+  upvoteAPIRequest,
+  downvoteAPIRequest,
+}: VoteHookProp) {
+  const { isAuthenticated, getIdToken } = useAuth();
+  const [idToken, setIdToken] = useState<string | undefined>(undefined);
+  const [upvotesLocal, setUpvotes] = useState<number>(upvotes);
+  const [downvotesLocal, setDownvotes] = useState<number>(downvotes);
+  const [hasUpVoted, setHasUpVoted] = useState<boolean>(false);
```
Comment (AaronCQL) [inline client/utils/hooks/useUpvoteDownvote.ts:37]: Consistent casing plox LOL

Think we just stick to using `downvote`/`upvote` as a single word to match BE bah

File: client/components/questions/UpvoteDownvote.tsx [hunk 2]
```diff
@@ -31,14 +27,23 @@ const QuestionUpvoteDownvote: FC<PageHeaderComponent> = ({
...
+    downvotesLocal,
+    upvoteOnClick,
+    downvoteOnClick,
+  } = useUpvoteDownvote({
+    upvotes,
+    downvotes,
+    qid,
+    checkVoteStatus: checkVoteQuestion,
+    upvoteAPIRequest: upvoteQuestion,
+    downvoteAPIRequest: downvoteQuestion,
```
Comment (EugeneTeu) [inline client/components/questions/UpvoteDownvote.tsx:44]: @AaronCQL when answers then will be `upvoteAPIRequest: upvoteAnswer`

**Comment by EugeneTeu** [comment]
> LGTM
> 
> 2 future points of consideration:
> 
> * Can make the `useUpvoteDownvote` hook support answers vote as well - so we can reuse it for answers
> * For the UI, can the we merge the counter display together with the vote button so that it's one thing? Right now a little confusing because there is a button to upvote and a display of the votes itself.

sure, but we intend to refactor the appearance anyways so later we adjust we do together bah

**Review by AaronCQL — APPROVED** [review]

**Pull request merged**